docs/roomHandlers/fishingRoom.md

```diff
@@ -0,0 +1,38 @@
+# Fishing Room
+
+A room with a beach and a boat dock. The player can fish from dock or leave. Fishing may result in catching a fish.
+
+## States
+
+- **AtEntrance**: The player is on a dock with a good/bad fishing pole. Total number of fish caught is X so far!
+- **Fishing**: Hopefully the fish are hungry today.
+- **Caught**: Congratulations!!! The player caught a fish!!!
+- **BrokenLine**: The player has a broken their fishing line and the fish got away.
+
+## Transitions
+
+- **leave**: The player leaves through the only available door. Total number of fish caught is X.
+- **fish**: The player casts their fishing line.
+- **fix**: The player fixs their fishing pole so they can fish again.
+- **stop**: The player stops fishing.
+
+## Diagram
+
+```mermaid
+stateDiagram-v2
+  [*] --> AtEntrance
+
+  AtEntrance --> Fishing: fish
```

Comment:
Should the `fish` command take you to `BrokenLine` when the rod is broken?

Reply:
Yes.

`AtEntrance` -> `Fishing` -> `Caught` OR `BrokenLine` -> `AtEntrance`

Reply:
I don't follow. If my comment is true then you would need to have

```
AtEntrance --> Fishing: fish (rod is working)
AtEntrance --> BrokenLine: fish (rod is broken)
```

because you have 

```
BrokenLine --> AtEntrance: stop
```

which implies you can leave BrokenLine without fixing the rod.

Reply:
Ahhh. Yeahhhh, so if the player chooses not to fix their broken fishing pole, they should go back to the `BrokenLine` state if they `fish`.